greedy_heuristics/scripts.py

```diff
@@ -0,0 +1,75 @@
+import csv
+import numpy as np
+import random
+import matplotlib.pyplot as plt
+
+
+def get_dist_matrix(path: str):
+    with open(path, 'r') as f:
+        reader = csv.reader(f, delimiter=';')
+        data = list(reader)
+
+    data = np.array(data).astype(int)
+
+    coords = data[:, :2]
+    costs = data[:, 2]
+
+    distances = np.round(np.sqrt(np.sum((coords[:, None, :] - coords[None, :, :]) ** 2, axis=-1))).astype(int)
+
+    return distances + costs
+
+
+def get_coords_n_costs(path: str):
+    with open(path, 'r') as f:
+        reader = csv.reader(f, delimiter=';')
+        data = list(reader)
+
+    data = np.array(data).astype(int)
+
+    coords = data[:, :2]
+    costs = data[:, 2]
+
+    return coords, costs
+
+
+def random_solution(matrix):
+    n =  matrix.shape[0] // 2
+
+    # I wonder whether it should be even simpler or it is ok to leave it in this form
+    sol = np.array(random.sample(range(matrix.shape[0]), n))
+    
+    cost = sum(matrix[sol[i-1], sol[i]] for i in range(n))
+    
+    return sol, cost
+
+
+def show_solution(path, solution):
+    coords, node_costs = get_coords_n_costs(path)
+    plt.scatter(coords[:, 0], coords[:, 1], c=node_costs, cmap='gray')
+
+    best_tour_coords = np.append(solution, solution[0])
+    plt.plot(coords[best_tour_coords, 0], coords[best_tour_coords, 1], 'r-')

```

Comment:
please indicate edge's weight as a color scale, similar to the nodes, instead of plain red all over the place

Reply:
But edges can be seen since they represent an euclidean distances. The assumption here is that it looks messy because of the random solution. Imo no point in adding the second indicator of length at this point.